torch/csrc/jit/ir/ir.cpp

```diff
@@ -1929,6 +1929,36 @@ at::ArrayRef<Value*> createTupleUnpack(Value* v) {
   return g.insertNode(g.createTupleUnpack(v))->outputs();
 }
 
+void inlineCallStackOfBlock(
+    Block* b,
+    std::unordered_map<InlinedCallStack*, InlinedCallStackPtr>& new_cs_entires,
+    Function* callee,
+    Node* to_replace,
+    c10::optional<ModuleInstanceInfo> m_info) {
+  for (auto n : b->nodes()) {
+    auto new_node_cs = n->callstack();
+
+    InlinedCallStack* raw_callstack_ptr =
+        new_node_cs ? new_node_cs->get() : nullptr;
+
+    if (!new_cs_entires.count(raw_callstack_ptr)) {
+      if (new_node_cs) {
+        new_cs_entires[raw_callstack_ptr] =
+            c10::make_intrusive<InlinedCallStack>(
+                *new_node_cs, callee, to_replace->sourceRange(), m_info);
+      } else {
+        new_cs_entires[raw_callstack_ptr] =
+            c10::make_intrusive<InlinedCallStack>(
+                callee, to_replace->sourceRange(), m_info);
+      }
+    }
```

Comment:
This seem to duplicate the code from `inlineCallTo` almost identically. Would it be possible to get rid of that duplication by calling `inlineCallStackOfBlock` there?

Reply:
Yes. I was almost gonna do that, but wanted to move too fast. Let me refactor it.

Reply:
> Also, why is it possible that !new_cs_entires.count(raw_callstack_ptr) is true but if (new_node_cs) is false?

So say you are inlining:
```
class B(..):
  def forward(self, x):
     return x + 3
class A(..):
  def forward(self, y):
    return self.b.forward(y)
```
Then when you are inlining A's forward method you replace
`%something = prim::CallMethod(...)` with `%something = aten::add(..`
This aten::add node is created by cloning B's forward method. At that time aten::add node's cs is null. And new_cs_entires may not have an entry for raw_callstack_ptr = nullptr so `if (!new_cs_entires.count(raw_callstack_ptr)) ` would be true.